library/Hasql/Pool/Observation.hs

```diff
@@ -0,0 +1,17 @@
+module Hasql.Pool.Observation where
+
+import Hasql.Pool.Prelude
+
+data Observation
+  = ConnectionEstablishedObservation UUID
+  | AttemptingToConnectObservation UUID
+  | FailedToConnectObservation UUID (Maybe ByteString)
+  | ConnectionReleasedObservation UUID ReleaseReason
```

Comment:
Q: Why does the UUID need to be generated internally? (just wondering if it can be left out for simplifying the code)

Reply:
It's for identifying the connection. Lets the user isolate events on a particular connection